### PR TITLE
Backport configurable map cursor interval

### PR DESCRIPTION
**Pull request by Warriorrrr**
Backports my previous PR #6385 to 1.16

### CODE REVIEW

File: Spigot-Server-Patches/0764-Backport-Configurable-item-frame-map-cursor-interval.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+diff --git a/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
+index 1df8fb8cb3fcf8201e1c5fa8ca13f7a9c632c379..d0d58dd33d3a1ae772f36eb8e0bc0953412f1d98 100644
+--- a/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
++++ b/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
+@@ -114,7 +114,7 @@ public class EntityTrackerEntry {
+             EntityItemFrame entityitemframe = (EntityItemFrame) this.tracker;
+             ItemStack itemstack = entityitemframe.getItem();
+ 
+-            if (this.tickCounter % 10 == 0 && itemstack.getItem() instanceof ItemWorldMap) { // CraftBukkit - Moved this.tickCounter % 10 logic here so item frames do not enter the other blocks
++            if (this.tracker.world.paperConfig.mapItemFrameCursorUpdateInterval >= 0 && this.tickCounter % this.tracker.world.paperConfig.mapItemFrameCursorUpdateInterval == 0 && itemstack.getItem() instanceof ItemWorldMap) { // CraftBukkit - Moved this.tickCounter % 10 logic here so item frames do not enter the other blocks // Paper - Make item frame map cursor update interval configurable
```
Comment (Machine-Maker) [inline Spigot-Server-Patches/0764-Backport-Configurable-item-frame-map-cursor-interval.patch]:
```suggestion
+            if (this.tracker.world.paperConfig.mapItemFrameCursorUpdateInterval > 0 && this.tickCounter % this.tracker.world.paperConfig.mapItemFrameCursorUpdateInterval == 0 && itemstack.getItem() instanceof ItemWorldMap) { // CraftBukkit - Moved this.tickCounter % 10 logic here so item frames do not enter the other blocks // Paper - Make item frame map cursor update interval configurable
```

Can't mod 0, thats a divide by 0 error. This has to be fixed on your original patch too (to 1.17).

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0764-Backport-Configurable-item-frame-map-cursor-interval.patch (hunk 1 above)
Comment (Warriorrrr) [reply Spigot-Server-Patches/0764-Backport-Configurable-item-frame-map-cursor-interval.patch]: Committed your suggestion, should I open a PR for the 1.17 one?

Comment (Machine-Maker) [reply Spigot-Server-Patches/0764-Backport-Configurable-item-frame-map-cursor-interval.patch]: yes please

**Review by Machine-Maker — APPROVED** [review]

**Comment by MiniDigger** [comment]
sorry for taking so long, but I am gonna close this PR.
1.16.x was in maintenance mode when you opened this and has since been moved to legacy status, as the vast majority of servers has updated.

**Pull request closed without merging**